File: docs/snippets/all/archetypes/gaussian_splats3d_ply.py
"""Log a 3D Gaussian Splatting (3DGS) PLY file."""

import sys

import rerun as rr

if len(sys.argv) < 2:
    print(f"Usage: {sys.argv[0]} <path_to_splats.ply>")
    sys.exit(1)

rr.init("rerun_example_gaussian_splats3d_ply", spawn=True)

rr.log_file_from_path(sys.argv[1])

File: docs/snippets/all/archetypes/gaussian_splats3d_ply.cpp
// Log a 3D Gaussian Splatting (3DGS) PLY file.

#include <rerun.hpp>

#include <iostream>

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cerr << "Usage: " << argv[0] << " <path_to_splats.ply>"
                  << std::endl;
        return 1;
    }

    const auto path = argv[1];

    const auto rec =
        rerun::RecordingStream("rerun_example_gaussian_splats3d_ply");
    rec.spawn().exit_on_failure();

    rec.log_file_from_path(path);
}
